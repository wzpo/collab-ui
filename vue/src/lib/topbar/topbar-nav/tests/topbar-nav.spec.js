import { shallowMount } from '@vue/test-utils';
import TopbarNav from '../index.vue';
import List from '../../../list/index.vue';
import Vue from 'vue';

Vue.component(List.name, List);

describe('TopbarNav', () => {
  it('should match snapshot', () => {
    const wrapper = shallowMount(TopbarNav, {
      propsData: {
        id: 'test'
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render one TopbarNav', () => {
    const container = shallowMount(TopbarNav);

    expect(container.classes('md-top-bar__nav')).toBeTruthy();
  });

  it('should add customized class name if className prop is set', () => {
    const container = shallowMount(TopbarNav, {
      propsData: {
        class: 'testClassName'
      }
    });

    expect(container.find('.testClassName').exists()).toBeTruthy();
  });

  it('should render children', () => {
    const container = shallowMount(TopbarNav, {
      slots: {
        default: `<div class="testingforTbN" />`
      }
    });

    expect(container.findAll('.testingforTbN').length).toEqual(1);
  });
});

