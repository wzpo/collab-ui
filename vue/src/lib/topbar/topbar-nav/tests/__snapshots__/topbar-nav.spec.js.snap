// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopbarNav should match snapshot 1`] = `
<div class="md-top-bar__nav" id="test">
  <md-list-stub focusfirst="true" focusfirstquery="" focusquery="" itemrole="listitem" role="list" shouldloop="true" tabtype="horizontal" trackactive="true"></md-list-stub>
</div>
`;
